### PR TITLE
New filter for rainbow, type or class focused units

### PR DESCRIPTION
**Pull request by AndreMantas**

### CODE REVIEW

File: src/app/shared/pipes/unit-filter.pipe.ts [hunk 1]
```diff
@@ -65,6 +67,22 @@ export class UnitFilterPipe implements PipeTransform {
           break;
       }
     }
+    if (arg.abilityTargetType) {
+      switch (arg.abilityTargetType) {
+        case 'crew':
+          filtered = filtered.filter(u => this.targetsCrew(u.lvl5Ability));
+          break;
+        case 'type':
+          filtered = filtered.filter(u => this.targetsTypes(u.lvl5Ability, u.type));
```
Comment (ThePieBandit) [inline src/app/shared/pipes/unit-filter.pipe.ts:76]: I think there are some abilities which target multiple types. Particularly, on the buff side

File: src/app/shared/pipes/unit-filter.pipe.ts [hunk 2]
```diff
@@ -65,6 +67,22 @@ export class UnitFilterPipe implements PipeTransform {
...
+          break;
+        case 'class':
+          filtered = filtered.filter(u => this.targetsClasses(u.lvl5Ability, u.class1, u.class2));
+          break;
+        default:
+          break;
```
Comment (ThePieBandit) [inline src/app/shared/pipes/unit-filter.pipe.ts:82]: if the default case is unexpected, include a log statement

**Review by ThePieBandit — COMMENTED** [review]
left some comments

File: src/app/shared/pipes/unit-filter.pipe.ts (hunk 1 above)
Comment (AndreMantas) [reply src/app/shared/pipes/unit-filter.pipe.ts]: that wouldn't be a problem but it got me wondering about abilities which buff other types/classes besides the unit own type/classes. i think it also wouldn't work for dual unit as it was. now i double checked and all cases seem to be working fine. thanks

**Review by ThePieBandit — APPROVED** [review]
LGTM

**Pull request merged**